apiserver/pgadminservice/pgadminimpl.go

```diff
@@ -0,0 +1,285 @@
+	"github.com/crunchydata/postgres-operator/config"
+	"github.com/crunchydata/postgres-operator/internal/pgadmin"
+	"github.com/crunchydata/postgres-operator/kubeapi"
+
+	log "github.com/sirupsen/logrus"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const pgAdminServiceSuffix = "-pgadmin"
+
+// CreatePgAdmin ...
+// pgo create pgadmin mycluster
+// pgo create pgadmin --selector=name=mycluster
+func CreatePgAdmin(request *msgs.CreatePgAdminRequest, ns, pgouser string) msgs.CreatePgAdminResponse {
+	var err error
+	resp := msgs.CreatePgAdminResponse{
+		Status:  msgs.Status{Code: msgs.Ok},
+		Results: []string{},
+	}
+
+	log.Debugf("createPgAdmin selector is [%s]", request.Selector)
+
+	// try to get the list of clusters. if there is an error, put it into the
+	// status and return
+	clusterList, err := getClusterList(request.Namespace, request.Args, request.Selector)
+	if err != nil {
+		resp.SetError(err.Error())
+		return resp
+	}
+
+	for _, cluster := range clusterList.Items {
+		log.Debugf("adding pgAdmin to cluster [%s]", cluster.Name)
+
+		// generate the pgtask, starting with spec
+		spec := crv1.PgtaskSpec{
+			Namespace:   cluster.Namespace,
+			Name:        fmt.Sprintf("%s-%s", config.LABEL_PGADMIN_TASK_ADD, cluster.Name),
+			TaskType:    crv1.PgtaskPgAdminAdd,
+			StorageSpec: cluster.Spec.PrimaryStorage,
+			Parameters: map[string]string{
+				config.LABEL_PGADMIN_TASK_CLUSTER: cluster.Name,
+			},
+		}
+
+		task := &crv1.Pgtask{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: spec.Name,
+				Labels: map[string]string{
+					config.LABEL_PG_CLUSTER:       cluster.Name,
+					config.LABEL_PGADMIN_TASK_ADD: "true",
+					config.LABEL_PGOUSER:          pgouser,
+				},
+			},
+			Spec: spec,
+		}
+
+		if err := kubeapi.Createpgtask(apiserver.RESTClient, task, cluster.Namespace); err != nil {
+			log.Error(err)
+			resp.SetError("error creating tasks for one or more clusters")
+			resp.Results = append(resp.Results, fmt.Sprintf("%s: error - %s", cluster.Name, err.Error()))
+			continue
+		} else {
+			resp.Results = append(resp.Results, fmt.Sprintf("%s pgAdmin addition scheduled", cluster.Name))
+		}
+	}
+
+	return resp
+}
+
+// DeletePgAdmin ...
+// pgo delete pgadmin mycluster
+// pgo delete pgadmin --selector=name=mycluster
+func DeletePgAdmin(request *msgs.DeletePgAdminRequest, ns string) msgs.DeletePgAdminResponse {
+	var err error
+	resp := msgs.DeletePgAdminResponse{
+		Status:  msgs.Status{Code: msgs.Ok},
+		Results: []string{},
+	}
+
+	log.Debugf("deletePgAdmin selector is [%s]", request.Selector)
+
+	// try to get the list of clusters. if there is an error, put it into the
+	// status and return
+	clusterList, err := getClusterList(request.Namespace, request.Args, request.Selector)
+	if err != nil {
+		resp.SetError(err.Error())
+		return resp
+	}
+
+	for _, cluster := range clusterList.Items {
+		log.Debugf("deleting pgAdmin from cluster [%s]", cluster.Name)
```

Comment:
As mentioned for the create function, a small check needs to be added to keep this from running against a cluster from an earlier Operator release that has yet to be upgraded. It should be similar to this (pulled from `apiserver/failoverservice/failoverimple.go`:
```
        // check if the current cluster is not upgraded to the deployed
	// Operator version. If not, do not allow the command to complete
	if cluster.Annotations[config.ANNOTATION_IS_UPGRADED] == config.ANNOTATIONS_FALSE {
		resp.Status.Code = msgs.Error
		resp.Status.Msg = cluster.Name + msgs.UpgradeError
		return response
	}
```